python/pysmurf/client/debug/smurf_noise.py

```diff
@@ -747,26 +747,42 @@ def get_iv_data(self, iv_data_filename, band, high_current_mode=False):
             Dictionary with IV information for band.
         """
         self.log(f'Extracting IV data from {iv_data_filename}')
+
+        # Load the data from file
         iv_data = np.load(iv_data_filename, allow_pickle=True).item()
-        iv_band_data = iv_data[band]
-        iv_high_current_mode = iv_data['high_current_mode']
-        for ch in iv_band_data:
-            v_bias = iv_band_data[ch]['v_bias']
-            if iv_high_current_mode and not high_current_mode:
-                iv_band_data[ch]['v_bias'] = v_bias*self._high_low_current_ratio
-            elif not iv_high_current_mode and high_current_mode:
-                iv_band_data[ch]['v_bias'] = v_bias/self._high_low_current_ratio
+
+        iv_band_data = {}
+        band = np.ravel(np.array([band]).astype(int))  # cast as array for loop
+
+        for b in band:
+            # Extract band specfic data
+            iv_band_data[b] = iv_data[b]
+            iv_high_current_mode = iv_data['high_current_mode']
+
+            for ch in iv_band_data[b]:
+                # Loop over channels and match IV and data high/low current
+                # mode status
+                v_bias = iv_band_data[b][ch]['v_bias']
+                if iv_high_current_mode and not high_current_mode:
+                    iv_band_data[b][ch]['v_bias'] = v_bias * \
+                        self._high_low_current_ratio
+                elif not iv_high_current_mode and high_current_mode:
+                    iv_band_data[b][ch]['v_bias'] = v_bias / \
+                        self._high_low_current_ratio
+
         return iv_band_data
 
 
-    def get_si_data(self, iv_band_data, ch):
+    def get_si_data(self, iv_band_data, b, ch):
```

Comment:
For consistency across pysmurf, I vote we use band instead of b.  Also, having all the bands' data in `iv_band_data` is confusing since I assume the band in the name meant it's for a band.  Since get_si_data and the other functions don't need to know the band, can the functions that call these functions instead just take the iv data for the specific band they're being called on?